service/common/src/main/java/org/apache/polaris/service/catalog/iceberg/IcebergCatalogHandler.java

```diff
@@ -1010,6 +1028,31 @@ public void renameView(RenameTableRequest request) {
     CatalogHandlers.renameView(viewCatalog, request);
   }
 
+  private @Nonnull LoadTableResponse filterResponseToSnapshots(
+      LoadTableResponse loadTableResponse, String snapshots) {
+    if (snapshots == null || snapshots.equalsIgnoreCase(SNAPSHOTS_ALL)) {
+      return loadTableResponse;
+    } else if (snapshots.equalsIgnoreCase(SNAPSHOTS_REFS)) {
+      TableMetadata metadata = loadTableResponse.tableMetadata();
+
+      Set<Long> referencedSnapshotIds =
+          metadata.refs().values().stream()
+              .map(SnapshotRef::snapshotId)
+              .collect(Collectors.toSet());
+
+      TableMetadata filteredMetadata =
+          metadata.removeSnapshotsIf(s -> referencedSnapshotIds.contains(s.snapshotId()));
```

Comment:
[doubt] I thought we want to return the snapshots that are being referenced ? should this be a 
```suggestion
      TableMetadata filteredMetadata =
          metadata.removeSnapshotsIf(s -> ! referencedSnapshotIds.contains(s.snapshotId()));
```

Reply:
The method is a bit confusing! This reads as `remove snapshots if a snapshot's ID is not in referencedSnapshotIds`